src/mono/wasm/Makefile

```diff
@@ -69,12 +68,15 @@ endif
 EMCC_DEBUG_FLAGS =-g -Os -s ASSERTIONS=1 -DENABLE_NETCORE=1 -DDEBUG=1
 EMCC_RELEASE_FLAGS=-Oz --llvm-opts 2 -DENABLE_NETCORE=1
 
+STRIP_CMD=&& $(EMSDK_PATH)/upstream/bin/wasm-opt --strip-dwarf $(NATIVE_BIN_DIR)/dotnet.wasm -o $(NATIVE_BIN_DIR)/dotnet.wasm
```

Comment:
From what I can read emscripten should already be stripping debug info on optimized builds without -g, is that not happening?

Can you try `-gseparate-dwarf` instead so we get the DWARF outside the main file?

Reply:
Ok looks like this doesn't happen today since we pass `--profiling-funcs`:  happen if https://github.com/emscripten-core/emscripten/blob/7c2d0936d09727dd371d846c1b7b718d73a64db2/tools/building.py#L448-L455

Reply:
Ah but according to this we should already be passing --strip-dwarf to wasm-opt: https://github.com/emscripten-core/emscripten/blob/7c2d0936d09727dd371d846c1b7b718d73a64db2/tools/building.py#L1523-L1536

Reply:
Those flags seem to behave differently that wasm-opt --strip-dwarf. With --strip-dwarf i get a final output size of 2346801, I can't get that size with any of -g<X> -gseparate-dwarf.

Reply:
What I meant is that according to that building.py L1523 emscripten should _already_ be running `wasm-opt --strip-dwarf` after linking. So we're seeing some size savings if we run it a second time?

Reply:
Yes.